modulus/models/fengwu/fengwu.py

```diff
@@ -0,0 +1,1270 @@
+        )
+
+        self.fuser = FuserLayer(
+            dim=embed_dim * 2,
+            input_resolution=(6, resolution[1][0], resolution[1][1]),
+            depth=6,
+            num_heads=num_heads[1],
+            window_size=window_size,
+            drop_path=drop_path_fuser,
+        )
+
+        self.decoder_surface = DecoderLayer(
+            img_size=img_size,
+            patch_size=patch_size,
+            out_chans=4,
+            dim=embed_dim,
+            output_resolution=resolution[0],
+            middle_resolution=resolution[1],
+            depth=2,
+            depth_middle=6,
+            num_heads=num_heads[:2],
+            window_size=window_size[1:],
+            drop_path=drop_path,
+        )
+        self.decoder_z = DecoderLayer(
+            img_size=img_size,
+            patch_size=patch_size,
+            out_chans=pressure_level,
+            dim=embed_dim,
+            output_resolution=resolution[0],
+            middle_resolution=resolution[1],
+            depth=2,
+            depth_middle=6,
+            num_heads=num_heads[:2],
+            window_size=window_size[1:],
+            drop_path=drop_path,
+        )
+        self.decoder_r = DecoderLayer(
+            img_size=img_size,
+            patch_size=patch_size,
+            out_chans=pressure_level,
+            dim=embed_dim,
+            output_resolution=resolution[0],
+            middle_resolution=resolution[1],
+            depth=2,
+            depth_middle=6,
+            num_heads=num_heads[:2],
+            window_size=window_size[1:],
+            drop_path=drop_path,
+        )
+        self.decoder_u = DecoderLayer(
+            img_size=img_size,
+            patch_size=patch_size,
+            out_chans=pressure_level,
+            dim=embed_dim,
+            output_resolution=resolution[0],
+            middle_resolution=resolution[1],
+            depth=2,
+            depth_middle=6,
+            num_heads=num_heads[:2],
+            window_size=window_size[1:],
+            drop_path=drop_path,
+        )
+        self.decoder_v = DecoderLayer(
+            img_size=img_size,
+            patch_size=patch_size,
+            out_chans=pressure_level,
+            dim=embed_dim,
+            output_resolution=resolution[0],
+            middle_resolution=resolution[1],
+            depth=2,
+            depth_middle=6,
+            num_heads=num_heads[:2],
+            window_size=window_size[1:],
+            drop_path=drop_path,
+        )
+        self.decoder_t = DecoderLayer(
+            img_size=img_size,
+            patch_size=patch_size,
+            out_chans=pressure_level,
+            dim=embed_dim,
+            output_resolution=resolution[0],
+            middle_resolution=resolution[1],
+            depth=2,
+            depth_middle=6,
+            num_heads=num_heads[:2],
+            window_size=window_size[1:],
+            drop_path=drop_path,
+        )
+
+    def forward(self, surface, z, r, u, v, t):
```

Comment:
I think we should make it principle to have the forward call be strictly `def forward(self, x):` and the internal of the forward model separates the components as necessary.

Reply:
Changed the forward to be one input. How about the output format?